Add info to clarify how to use the secrets for credentials into the ci-operator jobs (#539)
openshift/ci-docs: content/en/docs/architecture/ci-operator.md

      - name: "pipeline:bin" # the original definition of ${DEP}
        env: "DEP"
{{< / highlight >}}

### Using secrets to retrieve credentials

If your job requires to authenticate to a service,
for example to pull an image from a registry, 

> you might have the configuration available in the build cluster already.

Reviewer: ```suggestion
the required credentials might already be available in the build cluster.
```

For example, if you need to authenticate to the `Red Hat` registry (`registry.redhat.io`):

{{< highlight yaml >}}
- as: my-example-job
  steps:
    test:
      - as: my-example-job
        commands: |
          REGISTRY_AUTH_FILE=/var/run/secrets/ci-pull-credentials/.dockerconfigjson make test
        credentials:
          - mount_path: /var/run/secrets/ci-pull-credentials
            name: ci-pull-credentials
            namespace: ci
        from: src
        resources:
          requests:
            cpu: 100m
{{< / highlight >}}

> 

Reviewer: ```suggestion
This mounts the `ci-pull-credentials` secret into the test container and sets the path to the Docker config file so that registry authentication works as expected.

```

      - name: "pipeline:bin" # the original definition of ${DEP}
        env: "DEP"
{{< / highlight >}}

### Using secrets to retrieve credentials

If your job requires to authenticate to a service,
you can use the `secrets` that contains the credentials.

> Note that for some services, you might have the configuration

Author: Have you better wording suggestions?

available in the build cluster already.

For example, if you need to authenticate to the `Red Hat` registry (`registry.redhat.io`):

{{< highlight yaml >}}
- as: my-example-job
  steps:
    test:
      - as: my-example-job
        commands: |
          REGISTRY_AUTH_FILE=/var/run/secrets/ci-pull-credentials/.dockerconfigjson make test
        credentials:
          - mount_path: /var/run/secrets/ci-pull-credentials
            name: ci-pull-credentials
            namespace: ci
        from: src
        resources:
          requests:
            cpu: 100m
{{< / highlight >}}

> You can check the content of the secret in the `ci` namespace as the registries

Author: Fixed 👍
